pkg/apis/pipeline/v1beta1/task_validation.go

```diff
@@ -612,8 +612,7 @@ func isParamRefs(s string) bool {
 // - `trParams` are params from taskrun.
 // - `taskSpec` contains params declarations.
 func (ts *TaskSpec) ValidateParamArrayIndex(ctx context.Context, params Params) error {
-	cfg := config.FromContextOrDefaults(ctx)
-	if cfg.FeatureFlags.EnableAPIFields != config.AlphaAPIFields {
+	if !config.CheckAlphaOrBetaAPIFields(ctx) {
```

Comment:
This should be removed, since setting "stable" for v1beta1 also enables beta features

Reply:
Oh 🤯 
how about v1? Should we remove it from v1?

Reply:
In v1 we do check for beta API fields, so it makes sense to keep in v1

Reply:
Thanks!